pilot/platform/cloudfoundry/servicediscovery.go

```diff
@@ -104,9 +103,30 @@ func (sd *ServiceDiscovery) Instances(hostname string, ports []string,
 	return instances, nil
 }
 
-// HostInstances is not currently implemented for Cloud Foundry
 func (sd *ServiceDiscovery) HostInstances(addrs map[string]bool) ([]*model.ServiceInstance, error) {
-	return nil, nil
+	resp, err := sd.Client.Routes(context.Background(), new(copilotapi.RoutesRequest))
+	if err != nil {
+		return nil, fmt.Errorf("getting host instances: %s", err)
+	}
+
+	var instances []*model.ServiceInstance
+
+	for hostname, backendSet := range resp.GetBackends() {
+		service := newService(hostname)
+
+		for _, backend := range backendSet.GetBackends() {
+			instances = append(instances, &model.ServiceInstance{
+				Endpoint: model.NetworkEndpoint{
+					Address:     backend.Address,
+					Port:        int(backend.Port),
+					ServicePort: service.Ports[0],
+				},
+				Service: service,
```

Comment:
I am not an expert in Go, but I thought our code used this as a pointer thing.. instance.service.x/y/b.. @ZackButcher / @ijsnellf

Reply:
It LGTM - `newService` returns a pointer to a `model.Service`. Go checks this stuff at compile time (`*foo` is a different type than `foo`), so we're safe on that front.